Services/Diagnostics/DiagnosticsLogger.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Http;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Runtime;
+
+namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Diagnostics
+{
+    public interface IDiagnosticsLogger
+    {
+        Task<IHttpResponse> LogServiceStartAsync(string message);
+
+        Task<IHttpResponse> LogServiceHeartbeatAsync();
+
+        Task<IHttpResponse> LogServiceErrorAsync(
+            string message,
+            string exceptionMessage = "",
+            [CallerMemberName] string callerName = "",
+            [CallerFilePath] string filePath = "",
+            [CallerLineNumber] int lineNumber = 0);
+
+        Task<IHttpResponse> LogServiceErrorAsync(
+            string message,
+            object data,
+            [CallerMemberName] string callerName = "",
+            [CallerFilePath] string filePath = "",
+            [CallerLineNumber] int lineNumber = 0);        
+    }
+
+    public class DiagnosticsLogger : IDiagnosticsLogger
+    {
+        public struct JsonStruct
+        {
+            public string EventType;
+            public Dictionary<string, object> EventProperties;
```

Comment:
to make sure these two properties are serialized correctly when sent out, pls add `[JsonProperty(PropertyName = "...")]` on each